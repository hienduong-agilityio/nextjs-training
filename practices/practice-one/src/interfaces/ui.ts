export interface IInputGroupProps {
  placeholder?: string;
  value?: string;
  buttonText?: string;
  inputName?: string;
  isDisabled?: boolean;
  isLoading?: boolean;
  startIcon?: React.ReactNode;
  customClass?: {
    container?: string;
    inputContainer?: string;
    input?: string;
    button?: string;
  };
  onSubmit?: (event: React.FormEvent<HTMLFormElement>) => void;
}
