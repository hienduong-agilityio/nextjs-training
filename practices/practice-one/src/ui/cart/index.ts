export * from './CartTable';
export * from './OrderSummary';
export * from './VoucherInput';
export * from './EmptyCart';

export { default as CartItemRow } from './CartItemRow';
