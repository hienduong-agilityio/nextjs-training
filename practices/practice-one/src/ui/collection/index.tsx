export * from './LoadingCategoryGroup';
export * from './LoadingFilterSortBar';
