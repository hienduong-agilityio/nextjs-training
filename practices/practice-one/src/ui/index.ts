export * from './products/index';
export * from './collection/index';
