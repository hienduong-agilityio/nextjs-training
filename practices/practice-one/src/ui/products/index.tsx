export * from './CartAndFavoriteActions';
export * from './ProductImages';
export * from './ProductInfo';
export * from './QuantityControl';
export * from './SocialShare';
export * from './RelatedProduct';
export * from './ProductDetailTabs';
export * from './ProductReview';
export * from './ProductActions';
export * from './ProductLoading';
