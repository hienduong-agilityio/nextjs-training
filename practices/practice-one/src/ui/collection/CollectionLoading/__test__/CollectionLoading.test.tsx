import { render, screen } from '@testing-library/react';

// UI
import { LoadingCategoryGroup, LoadingFilterSortBar } from '@/ui';

describe('Loading Components for Collection', () => {
  it('renders LoadingCategoryGroup and matches snapshot', () => {
    const { asFragment } = render(<LoadingCategoryGroup />);
    expect(screen.getByLabelText('Loading Category Group')).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders LoadingFilterSortBar and matches snapshot', () => {
    const { asFragment } = render(<LoadingFilterSortBar />);
    expect(
      screen.getByLabelText('Loading Filter Sort Bar'),
    ).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });
});
