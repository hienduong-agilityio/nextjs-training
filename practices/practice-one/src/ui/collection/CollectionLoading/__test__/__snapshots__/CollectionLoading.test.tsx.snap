// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Components for Collection renders LoadingCategoryGroup and matches snapshot 1`] = `
<DocumentFragment>
  <div
    aria-label="Loading Category Group"
    class="mb-6 bg-secondary-100 p-4 rounded"
  >
    <div
      class="w-3/4 h-8 bg-gray-200 animate-pulse mb-4"
    />
    <ul
      class="space-y-2"
    >
      <div
        class="w-full h-6 bg-gray-200 animate-pulse "
      />
      <div
        class="w-full h-6 bg-gray-200 animate-pulse "
      />
      <div
        class="w-full h-6 bg-gray-200 animate-pulse "
      />
      <div
        class="w-full h-6 bg-gray-200 animate-pulse "
      />
      <div
        class="w-full h-6 bg-gray-200 animate-pulse "
      />
    </ul>
  </div>
</DocumentFragment>
`;

exports[`Loading Components for Collection renders LoadingFilterSortBar and matches snapshot 1`] = `
<DocumentFragment>
  <div
    aria-label="Loading Filter Sort Bar"
    class="bg-secondary-100 p-6 shadow-md md:sticky rounded top-0 z-50 w-full"
  >
    <div
      class="flex flex-col sm:flex-row sm:justify-between sm:items-center space-y-4 sm:space-y-0"
    >
      <div
        class="w-20 h-6 bg-gray-200 animate-pulse"
      />
      <ul
        class="flex flex-col sm:flex-row sm:items-center sm:space-x-4 space-y-4 sm:space-y-0"
      >
        <div
          class="w-20 h-6 bg-gray-200 animate-pulse"
        />
        <div
          class="w-20 h-6 bg-gray-200 animate-pulse"
        />
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
